programs/lib/ctx_switch.S

```diff
@@ -0,0 +1,137 @@
+/**
+ * FlexPRET's context switch mechanism
+ *
+ */
+
+#include "flexpret_csrs.h"
+
+.section .text
+.globl thread_ctx_switch_store
+.type thread_ctx_switch_store,@function
+
+/**
+ * The contexts variable is declared as
+ *
+ * struct thread_ctx_t contexts[NUM_THREADS];
+ *
+ * where the thread_ctx_t is aligned to 32 words for the sake of simplicity.
+ * Getting the context for a specific thread is then calculating the offset;
+ * see the macro itself for details.
+ */
+.macro getContexts
+    // Get a0 as the address of the contexts variable
```

Comment:
Would be nice with a comment to understand why you need to change the assembly options before loading contexts into a0

Reply:
Haha to be honest I don't know that myself; I just took the code from start.S. But I found an explanation that made sense online :)